crates/core_arch/src/x86/avx512f.rs

```diff
@@ -13441,6 +13587,54 @@ pub unsafe fn _mm512_mask_i32gather_ps(
     transmute(r)
 }
 
+/// Gather single-precision (32-bit) floating-point elements from memory using 32-bit indices. 32-bit elements are loaded from addresses starting at base_addr and offset by each 32-bit element in vindex (each index is scaled by the factor in scale). Gathered elements are merged into dst using writemask k (elements are copied from src when the corresponding mask bit is not set). scale should be 1, 2, 4 or 8.
+///
+/// [Intel's documentation](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#text=_mm256_mmask_i32gather_ps&expand=3009)
+#[inline]
+#[target_feature(enable = "avx512f,avx512vl")]
+#[cfg_attr(test, assert_instr(vgatherdps, scale = 1))]
+#[rustc_args_required_const(4)]
+pub unsafe fn _mm256_mmask_i32gather_ps(
+    src: __m256,
+    mask: __mmask8,
+    offsets: __m256i,
+    slice: *const u8,
+    scale: i32,
+) -> __m256 {
+    let slice = slice as *const f32;
+    macro_rules! call {
+        ($imm8:expr) => {
+            _mm256_i32gather_ps(slice, offsets, $imm8)
+        };
+    }
+    let r = constify_imm8_gather!(scale, call);
+    transmute(simd_select_bitmask(mask, r.as_f32x8(), src.as_f32x8()))
```

Comment:
`simd_select_bitmask` is incorrect in this case. We need to use the special LLVM intrinsic `llvm.x86.avx512.mask.gather3siv4.sf`.

The difference here is that the LLVM intrinsic will only access memory addresses which are not masked out. However your version will access memory addresses that are masked out, which could cause a crash if a masked value is an invalid offset.